### DISCUSSION TITLE
How to use .nml file or how to Adjust Mode Parameters

### DISCUSSION THREAD

**Discussion by fakaaaaaaaa12** [NOAA-EMC/WW3#1155] [Q&A]
Hi,
I am trying to perform wind driven WW3 calculations with an unstructured mesh and the results are not very good, so I wish to adjust the parameters in the scheme, and have the following questions:
1. I used ST4 switch for the compilation, do I need to recompile if I wish to change the parameter scheme?
2. If I don't need to recompile, but only adjust the value of a parameter, do I need to use the nml file? Do all nml files have to be prepared in my working directory? (For example, can I prepare only ww3_grid.nml?)
3. If I prepare only one ww3_grid.nml, what should be the exact format (although there is an example, I would like to have a more specific reference)
See attachment for my grid.inp
[ww3_grid.inp.txt](https://github.com/NOAA-EMC/WW3/files/13769753/ww3_grid.inp.txt)

**Comment by JessicaMeixner-NOAA** [comment]
You only have to recompile the code if you change you want to change your switch file. Otherwise, you can use the same executables even if you change parameters in ww3_grid.inp or ww3_grid.nml.     There are many options of how to configure WW3, there is not only one way to prepare ww3_grid.nml.  For exmaples, I'd recommend looking at the regression tests to see how various tests are configured.

**Comment by aronroland** [comment]
Hi, 

 

if u want us to take a look into this case we need the full setup. Morever, please clarify „results are not very good“. 

 

Cheers

 

Aron

 

Von: fakaaaaaaaa12 ***@***.***> 
Gesendet: Dienstag, 26. Dezember 2023 08:59
An: NOAA-EMC/WW3 ***@***.***>
Cc: Subscribed ***@***.***>
Betreff: [NOAA-EMC/WW3] How to use .nml file or how to Adjust Mode Parameters (Discussion #1155)

 

Hi,
I am trying to perform wind driven WW3 calculations with an unstructured mesh and the results are not very good, so I wish to adjust the parameters in the scheme, and have the following questions:

1.	I used ST4 switch for the compilation, do I need to recompile if I wish to change the parameter scheme?
2.	If I don't need to recompile, but only adjust the value of a parameter, do I need to use the nml file? Do all nml files have to be prepared in my working directory? (For example, can I prepare only ww3_grid.nml?)
3.	If I prepare only one ww3_grid.nml, what should be the exact format (although there is an example, I would like to have a more specific reference)
See attachment for my grid.inp
ww3_grid.inp.txt <https://github.com/NOAA-EMC/WW3/files/13769753/ww3_grid.inp.txt>
